settings.gradle

```diff
@@ -123,15 +39,177 @@ include 'lowcodelanguage'
 include 'fmlearlydisplay'
 include 'forge-transformers'
 
-include ':mcp'
-project(":mcp").projectDir = file("projects/mcp")
+enableFeaturePreview 'TYPESAFE_PROJECT_ACCESSORS'
+
+gradle.beforeProject { Project project ->
+    project.pluginManager.withPlugin('net.minecraftforge.forge.build.convention') {
+        // NOTE: We are defining these variables in here so that the projects can stay isolated, increasing build time.
+        // See minecraft.versions.toml for the versions where you can change these values.
+        //@formatter:off
+        project.ext.javaVersion          = project.gradleutils.unpack(project.bootLibs.versions.java)
+        project.ext.minecraftVersion     = project.gradleutils.unpack(project.bootLibs.versions.minecraft)
+        project.version                  = project.gitversion.getMCTagOffsetBranch(project.ext.minecraftVersion)
+        project.ext.forgeVersion         = project.version.substring(project.ext.minecraftVersion.length() + 1)
+        project.ext.minecraftNextVersion = project.gradleutils.unpack(project.bootLibs.versions.minecraft.next)
+        project.ext.mcpVersion           = project.gradleutils.unpack(project.bootLibs.versions.mcp)
+        project.ext.mappingsChannel      = project.gradleutils.unpack(project.bootLibs.versions.mappings.channel)
+        project.ext.mappingsVersion      = project.gradleutils.unpack(project.bootLibs.versions.mappings.version)
+        //@formatter:on
+    }
+
+    // NOTE: We are adding dependencies to projects instead of through settings, so we don't have to apply forgedev in settings.
+    // Applying forgedev in settings will break IDE linting support, which makes it very hard to work with.
+    // We can reconsider this if/when switching to Kotlin DSL, or if JetBrains gets their shit together.
+    project.repositories {
+        mavenCentral()
+        maven { url = 'https://libraries.minecraft.net/' }
+
+        project.pluginManager.withPlugin('net.minecraftforge.gradleutils') {
+            maven project.gradleutils.forgeMaven
+        }
+
+        project.pluginManager.withPlugin('net.minecraftforge.forgedev') {
+            maven project.forgedev.mavenizer
+        }
+
+        mavenLocal()
+    }
+}
+
+dependencyResolutionManagement.versionCatalogs {
+    register('bootLibs') {
```

Comment:
I would rename this to something like variables or something. These really arnt bootLibs.
And then we should have a section called bootLibs that ist he old `bootstrap` config. So that we know what files need to be on the boot classpath.